src/System.Collections.Specialized/tests/NameObjectCollectionBase/KeysCollectionCopyToTests.cs

```diff
@@ -258,9 +255,9 @@ void CheckArray(Array array, ArrayList expected)
             }
             for (int i = 0; i < expected.Count; i++)
             {
-                if (((String[])array)[i] != (String)expected[i])
+                if (((string[])array)[i] != (string)expected[i])
                 {
-                    Assert.False(true, string.Format("Value {0} is incorrect.  array[{0}]={1}, should be {2}", i, ((String[])array)[i], (String)expected[i]));
+                    Assert.False(true, string.Format("Value {0} is incorrect.  array[{0}]={1}, should be {2}", i, ((string[])array)[i], (string)expected[i]));
```

Comment:
As long as you're changing these lines, shouldn't this just be an Assert.NotSame check?